### DISCUSSION TITLE
Issue with php stan

### DISCUSSION THREAD

**Discussion by vico-gs** [spatie/image#169] [Q&A]
Hey,

I have the following code that changes the background and optimize my image:

`
        Image::load($filePathName)->background('transparent')->optimize()->save();
`

It works pretty well but, when I use phpstan on level 2, I get the following error:

` Call to an undefined method Spatie\Image\Manipulations::save()`

Despite the fact that the method does exist...
Does anyone faced the same issue ?

**Comment by freekmurze** [comment]
Feel free to PR improvements

**Reply by parallels999** [reply]
https://github.com/spatie/image/pull/198

**Comment by vico-gs** [comment]
Alright I figured out where the problem come from. The __call method from the class Image is used to return the object everytime a method is used. I think that the issue comes from php stan that doesn't recognize the type returned by the function __call, so phpstan doesn't understand that the save method does exists and returned the error of an undefined method. So the practice work but the php stan scan doesn't on this particular case.

**Reply by Eliepse** [reply]
In deed! This probably is a PHPStan problem then. Maybe worth talking about this issue over the PHPStan project?

**Reply by vico-gs** [reply]
Yes ! Thanks will do !

**Comment by michellaurent** [comment]
I fixed that by using the `manipulate()`method as [written in the docs](https://spatie.be/docs/image/v1/usage/preparing-manipulations ).

```php
$manipulations = (new Manipulations())->manualCrop($this->crop['width'], $this->crop['height'], $this->crop['x'], $this->crop['y']);

Image::load(storage_path('app/public/'.$picture))->manipulate($manipulations)->save();
```

**Reply by websitevirtuoso** [reply]
```
Image::load($tmpFilePath)->format(Manipulations::FORMAT_WEBP)
            ->quality(100)->optimize()->width(100)->height(100)->save();
```
            
I have the same problem. It has to be fixed in package
            
Is very easy fix. for example this code will work
```
        Image::load($tmpFilePath)->save();
```
        
Problem that each method thould return $this (class itself) to use chain option. If you try to do this it will be fixed. for me I will just ignore this error via
         // @phpstan-ignore-next-line We know error in package
